IdentityCore/tests/integration/ios/MSIDDefaultBrokerResponseHandlerTests.m

```diff
@@ -0,0 +1,488 @@
+    
+    NSError *error = nil;
+    MSIDTokenResult *result = [brokerResponseHandler handleBrokerResponseWithURL:brokerResponseURL error:&error];
+    
+    XCTAssertNil(result);
+    XCTAssertNotNil(error);
+    XCTAssertEqual(error.code, 213);
+    XCTAssertEqualObjects(error.domain, @"MSALErrorDomain");
+    XCTAssertEqualObjects(error.userInfo[MSIDCorrelationIdKey], correlationId);
+    XCTAssertEqualObjects(error.userInfo[MSIDBrokerVersionKey], @"1.0.0");
+    XCTAssertEqualObjects(error.userInfo[MSIDHTTPResponseCodeKey], @200);
+    XCTAssertEqualObjects(error.userInfo[MSIDErrorDescriptionKey], @"AADSTS53005: Application needs to enforce intune protection policies");
+    XCTAssertEqualObjects(error.userInfo[MSIDOAuthErrorKey], @"unauthorized_client");
+    XCTAssertEqualObjects(error.userInfo[MSIDOAuthSubErrorKey], @"protection_policies_required");
+    XCTAssertEqualObjects(error.userInfo[MSIDUserDisplayableIdkey], @"user@contoso.com");
+    XCTAssertNil(error.userInfo[MSIDDeclinedScopesKey]);
+    XCTAssertNil(error.userInfo[MSIDGrantedScopesKey]);
+}
+
+- (void)testHandleBrokerResponse_whenBrokerIntuneErrorResponse_withAdditionalToken_shouldReturnNilResultAndError_andCacheMAMToken
+{
+    [self saveResumeStateWithAuthority:@"https://login.microsoftonline.com/common"];
+    
+    NSString *correlationId = [[NSUUID UUID] UUIDString];
+    
+    NSString *mamIdTokenString = [MSIDTestIdTokenUtil idTokenWithPreferredUsername:@"user@contoso.com"
+                                                                           subject:@"mysubject"
+                                                                         givenName:@"myGivenName"
+                                                                        familyName:@"myFamilyName"
+                                                                              name:@"Contoso"
+                                                                           version:@"2.0"
+                                                                               tid:@"contoso.com-guid"];
+    
+    NSDictionary *mamClientInfo = @{ @"uid" : @"1", @"utid" : @"1234-5678-90abcdefg"};
+    NSString *mamRawClientInfo = [mamClientInfo msidBase64UrlJson];
+    
+    NSDate *expiresOn = [NSDate dateWithTimeIntervalSinceNow:3600];
+    NSString *expiresOnString = [NSString stringWithFormat:@"%ld", (long)[expiresOn timeIntervalSince1970]];
+    
+    NSDate *extExpiresOn = [NSDate dateWithTimeIntervalSinceNow:36000];
+    NSString *extExpiresOnString = [NSString stringWithFormat:@"%ld", (long)[extExpiresOn timeIntervalSince1970]];
+    
+    NSString *scopes = @"myscope1 myscope2";
+    
+    NSDictionary *intuneMAMTokenDictionary =
+    @{
+      @"authority" : @"https://login.microsoftonline.com/common",
+      @"scope" : scopes,
+      @"client_id" : @"my_client_id",
+      @"id_token" : mamIdTokenString,
+      @"client_info" : mamRawClientInfo,
```

Comment:
Should we also return homeAccountId, as it might be not based on client_info in some cases in the future, and client_info is also optional in schema.

Reply:
Made changes in the protocol doc to have "home_account_id" being returned from broker. 
Do we want to change the current common core logic to fetch home_account_id field from server; if unavailable, then fallback to client info?